torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -623,6 +640,13 @@ void ProcessGroupNCCL::workCleanupLoop() {
         ++it;
       }
     }
+
+    if (workList_.empty()) {
+      // Notify the main thread if it is blocked in the shutdown sequence,
+      // waiting for the work vector to become empty.
+      lock.unlock();
+      workVectorCV_.notify_one();
```

Comment:
Do we really need to do this? Wouldn't this automatically abort when `terminateProcessGroup_` is set to True? Or are we referring to some other thread here?

Reply:
This notifies the CV in the destructor that is waiting for the workList_ to become empty.